modules/core/04-channel/v2/types/packet.go

```diff
@@ -2,10 +2,62 @@ package types
 
 import (
 	"crypto/sha256"
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	host "github.com/cosmos/ibc-go/v9/modules/core/24-host"
 )
 
+func (p Packet) ValidateBasic() error {
+	if len(p.Data) != 1 {
+		return errorsmod.Wrapf(ErrInvalidPacket, "packet data length must be 1, got %d", len(p.Data))
+	}
+
+	for _, pd := range p.Data {
```

Comment:
This would loop only once given the check above. I think it still looks nicer that a bunch of `p.Data[0]` but happy to change

Reply:
the loop is much better indeed!